XmlFormatter/src/Manager/VersionManager.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Octokit;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using XmlFormatter.src.DataContainer;
+using XmlFormatter.src.EventMessages;
+
+namespace XmlFormatter.src.Manager
+{
+    class VersionManager
+    {
+        public event EventHandler<BaseEventArgs> Error;
+
+        /// <summary>
+        /// This method will return you the current version of the application
+        /// </summary>
+        /// <returns>The current version of the application</returns>
+        public Version GetApplicationVersion()
+        {
+            Assembly assembly = Assembly.GetExecutingAssembly();
+            string versionString = "0.0.0";
+            using (Stream stream = assembly.GetManifestResourceStream("XmlFormatter.Version.txt"))
+            {
+                using (StreamReader reader = new StreamReader(stream))
+                {
+                    versionString = reader.ReadLine();
+                }
+            }
+
+            return ConvertInnerFormatToProperVersion(versionString);
+        }
+
+        /// <summary>
+        /// This method will convert a inner string version to a proper version instance
+        /// </summary>
+        /// <param name="stringVersion">The string version to use for converting</param>
+        /// <returns>A proper version class</returns>
+        public Version ConvertInnerFormatToProperVersion(string stringVersion)
+        {
+            stringVersion += ".0";
+            Regex regex = new Regex(@"[0-9]{1,}.[0-9]{1,}.[0-9]{1,}");
+            if (!regex.IsMatch(stringVersion))
+            {
+                stringVersion = "0.0.0.0";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'stringVersion'.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5002975)